sympy/polys/tests/test_hypothesis.py

```diff
@@ -1,15 +1,27 @@
-from hypothesis import given
+from hypothesis import given, assume
 from hypothesis import strategies as st
 from sympy.abc import x
 from sympy.polys.polytools import Poly
+from sympy.polys.domains import ZZ
+
+
+@st.composite
+def coefficients(draw: st.DrawFn, empty=True):
+    l = [ZZ(i) for i in draw(st.lists(st.integers()))]
+    if not empty:
+        assume(len(l) > 0)
+    if len(l) > 0:
+        assume(any(l))
+        assume(l[0] != 0)
```

Comment:
What are we trying to do there? The first assumption says any of the elements can't be zero, and the second says only the first one can't be zero. I would guess you want the latter

```suggestion
        assume(l[0] != 0)
```